src/ServerDaemon.chpl

```diff
@@ -0,0 +1,510 @@
+
+        :arg token: the submitted token string
+        */
+        proc authenticateUser(token : string) throws {
+            if token == 'None' || token.isEmpty() {
+                throw new owned ErrorWithMsg("Error: access to arkouda requires a token");
+            }
+            else if serverToken != token {
+                throw new owned ErrorWithMsg("Error: token %s does not match server token, check with server owner".format(token));
+            }
+        }
+
+        /*
+        Sets the shutdownDaemon boolean to true and sends the shutdown command to socket,
+        which stops the arkouda_server listener thread and closes socket.
+        */
+        override proc shutdown(user: string) throws {
+            if saveUsedModules then
+                writeUsedModules();
+            this.shutdownDaemon = true;
+            this.repCount += 1;
+            this.socket.send(serialize(msg="shutdown server (%i req)".format(repCount), 
+                         msgType=MsgType.NORMAL,msgFormat=MsgFormat.STRING, user=user));
+        }
+
+        /**
+         * Register our server commands in the CommandMap
+         * There are 3 general types
+         * 1. Standard, required commands which adhere to the standard Message signature
+         * 2. Specialized, required commands which do not adhere to the standard Message signature
+         * 3. "Optional" modules which are included at compilation time via ServerModules.cfg
+         */
+        proc registerServerCommands() {
+            registerBinaryFunction("tondarray", tondarrayMsg);
+            registerFunction("create", createMsg);
+            registerFunction("delete", deleteMsg);
+            registerFunction("set", setMsg);
+            registerFunction("info", infoMsg);
+            registerFunction("str", strMsg);
+            registerFunction("repr", reprMsg);
+            registerFunction("getconfig", getconfigMsg);
+            registerFunction("getmemused", getmemusedMsg);
+            registerFunction("getCmdMap", getCommandMapMsg);
+            registerFunction("clear", clearMsg);
+            registerFunction("lsany", lsAnyMsg);
+            registerFunction("readany", readAnyMsg);
+            registerFunction("getfiletype", getFileTypeMsg);
+
+            // For a few specialized cmds we're going to add dummy functions, so they
+            // get added to the client listing of available commands. They will be
+            // intercepted in the cmd processing select statement and processed specially
+            registerFunction("array", akMsgSign);
+            registerFunction("connect", akMsgSign);
+            registerFunction("disconnect", akMsgSign);
+            registerFunction("noop", akMsgSign);
+            registerFunction("ruok", akMsgSign);
+            registerFunction("shutdown", akMsgSign);
+
+            // Add the dynamic Modules/cmds implemented via ServerRegistration.chpl & ServerModules.cfg
+            doRegister();
```

Comment:
I know this was put up before the change but just wanted to note that this is no longer needed as of #1606